configstack/module.go

```diff
@@ -52,7 +56,86 @@ func ResolveTerraformModules(terragruntConfigPaths []string, terragruntOptions *
 		return []*TerraformModule{}, err
 	}
 
-	return crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	crossLinkedModules, err := crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	finalModules, err := flagExcludedDirs(crossLinkedModules, terragruntOptions)
+	if err != nil {
+		return []*TerraformModule{}, err
+	}
+
+	return finalModules, nil
+}
+
+//flagExcludedDirs iterates over a module slice and flags all entries as excluded, which should be ignored via the terragrunt-exclude-dir CLI flag.
+func flagExcludedDirs(modules []*TerraformModule, terragruntOptions *options.TerragruntOptions) ([]*TerraformModule, error) {
+
+	// Save original working directory
+	workingDir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+
+	// If no ExcludeDirs is specified return the modules list instantly
+	if len(terragruntOptions.ExcludeDirs) == 0 {
+		return modules, nil
+	}
+
+	excludeGlobMatches := []string{}
+
+	// If possible, expand the glob to get all excluded filepaths
+	// Change to the actual working dir to make glob expansion work correctly
+	os.Chdir(terragruntOptions.WorkingDir)
```

Comment:
Hm, I'm a bit worried about changing dirs... This can have unintended and confusing side effects, especially when things are running concurrently (e.g., multiple unit tests). 

Perhaps instead you should `filepath.Join` each excluded dir with `terragruntOptions.WorkingDir`if that excluded dir isn't an absolute path?

Reply:
I had the same idea, but I thought there were some reasons why it would not work. Let me think this through again and give it another try

Reply:
yep this worked, thanks for challenging this again!